android/src/main/java/com/applovin/reactnative/AppLovinMAXModule.java

```diff
@@ -2673,6 +2673,76 @@ private WritableMap createNetworkResponseInfo(final MaxNetworkResponseInfo respo
         return networkResponseObject;
     }
 
+    // Amazon
+
+    public void setAmazonBannerResult(final Object result, final String adUnitId)
+    {
+        setAmazonResult( result, adUnitId, MaxAdFormat.BANNER );
+    }
+
+    public void setAmazonMRecResult(final Object result, final String adUnitId)
+    {
+        setAmazonResult( result, adUnitId, MaxAdFormat.MREC );
+    }
+
+    public void setAmazonInterstitialResult(final Object result, final String adUnitId)
+    {
+        setAmazonResult( result, adUnitId, MaxAdFormat.INTERSTITIAL );
+    }
+
+    private void setAmazonResult(final Object result, final String adUnitId, final MaxAdFormat adFormat)
+    {
+        if ( !isValidToProceedForAdUnitId( adUnitId, result ) ) return;
+
+        if ( adFormat == MaxAdFormat.INTERSTITIAL )
+        {
+            MaxInterstitialAd interstitial = retrieveInterstitial( adUnitId, "setAmazonResult" );
+            setAmazonResult( result, interstitial );
```

Comment:
Since we already know the format and type of object at this level, we could just unwrap this method call here so that we don't need to make a check again in `setAmazonResult( result, interstitial );`

So this method could be like

```
private void setAmazonResult(final Object result, final String adUnitId, final MaxAdFormat adFormat)
    {
        if ( !isValidToProceedForAdUnitId( adUnitId, result ) ) return;
        
        String key = localExtraParameterKeyForAmazonResult( result );

        if ( adFormat == MaxAdFormat.INTERSTITIAL )
        {
            MaxInterstitialAd interstitial = retrieveInterstitial( adUnitId, "setAmazonResult" );
            interstitial.setLocalExtraParameter( key, result );
        }
        else
        {
            MaxAdView adView = retrieveAdView( adUnitId, adFormat );
            adView.setLocalExtraParameter( key, result );
        }
    }

```

We could also do something like that for iOS (even though iOS has no type checks anyway)